### PR TITLE
Clamp SSR calculation that was leading to NaNs and silently dropped from Wandb

### LINKED ISSUE #693: SSR ensemble metric not getting logged to Wandb correctly
As part of ai2cm/full-model#2264 I noticed that SSR was not being logged to Wandb ([example](https://wandb.ai/ai2cm/ace/runs/r1shdbfb?nw=nwusertroya)). I went back to my previous training runs and found that only 9 out of ~60 variables were being reported on Wandb during inference ([example](https://wandb.ai/ai2cm/ace/runs/4ikfm6oj?nw=nwusertroya) and [example](https://wandb.ai/ai2cm/ace/runs/nskivquh?nw=nwusertroya)). Note it is always the same 9 variables being plotted. This has been happening since at least July 21st.

### PR DESCRIPTION
**Pull request by Arcomano1234**
`SSRBiasMetric.get()` produces NaN values when the unbiased MSE correction overshoots to negative values (common with small ensembles or few batches), causing SSR to silently drop from WandB logs. This PR clamps the unbiased MSE before taking sqrt and returns a defined value (-1) when skill is zero, and also fixes the diverging colormap check in `_EnsembleAggregator` which was comparing against variable names instead of metric names.

Changes:
-` fme.ace.aggregator.one_step.ensemble.SSRBiasMetric.get`: clamp `total_unbiased_mse` to min=0.0 before sqrt to avoid NaN, return -1 via torch.where when skill is zero
-` fme.ace.aggregator.one_step.ensemble._EnsembleAggregator._get_data`: fix diverging colormap check to use metric in `self._diverging_metrics` instead of key in `self._diverging_metrics`

- [x] Tests added

Resolves #693

### CODE REVIEW

File: configs/experiments/2025-11-15-ace2s-x-shield/test-config.yaml
Comment (mcgibbon) [inline configs/experiments/2025-11-15-ace2s-x-shield/test-config.yaml:1]: Should clean this file up (delete it and revert the sh script) before merge.

File: fme/ace/aggregator/inference/test_main.py [hunk 1]
```diff
@@ -132,28 +130,101 @@ def test_inference_evaluator_aggregator_ensemble():
...
     gen_data = BatchData.new_for_testing(
         names=["a", "b", "c"],
-        n_samples=1,
-        n_timesteps=n_timesteps,
+        n_samples=n_ensemble,
+        n_timesteps=n_forward_steps,
         img_shape=(nx, ny),
     )
-
-    target_data = target_data.broadcast_ensemble(n_ensemble=n_ensemble)
```
Comment (mcgibbon) [inline fme/ace/aggregator/inference/test_main.py:149]: Why did this test initialization logic get changed?

File: fme/ace/aggregator/one_step/ensemble.py [hunk 2]
```diff
@@ -134,8 +134,14 @@ def get(self) -> torch.Tensor:
         if self._total_unbiased_mse is None or self._total_variance is None:
             raise ValueError("No batches have been recorded.")
         spread = self._total_variance.sqrt()
-        skill = self._total_unbiased_mse.sqrt()
-        return spread / skill - 1
+        # Clamp to avoid NaN from sqrt of negative values. The unbiased MSE
```
Comment (mcgibbon) [inline fme/ace/aggregator/one_step/ensemble.py:137]: These changes make sense and seem good.

**Review by mcgibbon — COMMENTED** [review]
Mostly LGTM, but clean up the submission directory, revert any unnecessary changes to tests, and update the PR description.

File: fme/ace/aggregator/one_step/ensemble.py (hunk 2 above)
Comment (Arcomano1234) [reply fme/ace/aggregator/one_step/ensemble.py:137]: Yeah I agree and I believe the NaNs are the root of this whole problem

**Review by mcgibbon — APPROVED** [review]
Please update the PR title, otherwise LGTM
